### PR TITLE
Add a realm-level setting to enable FGAP to a realm

### LINKED ISSUE #34920: [FGAP] Add a realm-level setting to enable FGAP to a realm
FGAP should be enabled/disabled per realm basis. There needs to be a switch in the UI to support it.

### PR DESCRIPTION
**Pull request by vramik**
Closes #34920

### CODE REVIEW

**Comment by vramik** [comment]
@agagancarczyk would you please take a look from UI POV?

File: model/infinispan/src/main/java/org/keycloak/models/cache/infinispan/RealmAdapter.java [hunk 1]
```diff
@@ -1838,6 +1849,11 @@ private boolean featureAwareIsOrganizationsEnabled(boolean isOrganizationsEnable
         return isOrganizationsEnabled;
     }
 
+    private boolean featureAwareIsAdminPermissionsEnabled(boolean isAdminPermissionsEnabled) {
```
Comment (pedroigor) [inline model/infinispan/src/main/java/org/keycloak/models/cache/infinispan/RealmAdapter.java:1852]: Perhaps we can simplify other places using the same approach with something like:

```
private boolean featureEnabled(Feature feature, boolean isVerifiableCredentialsEnabled) {}
```

Not a blocker, I hope you see my point ...

File: server-spi-private/src/main/java/org/keycloak/models/utils/ModelToRepresentation.java [hunk 2]
```diff
@@ -405,6 +405,7 @@ public static RealmRepresentation toRepresentation(KeycloakSession session, Real
         rep.setResetPasswordAllowed(realm.isResetPasswordAllowed());
         rep.setEditUsernameAllowed(realm.isEditUsernameAllowed());
         rep.setOrganizationsEnabled(realm.isOrganizationsEnabled());
+        rep.setAdminPermissionsEnabled(realm.isAdminPermissionsEnabled());
```
Comment (pedroigor) [inline server-spi-private/src/main/java/org/keycloak/models/utils/ModelToRepresentation.java:408]: In this class, we should also update `org.keycloak.models.utils.ModelToRepresentation#REALM_EXCLUDED_ATTRIBUTES` to exclude the new attribute from the attributes map.

Btw, we did not do that for the `ORGANIZATION_ENABLED` attribute :)

**Comment by pedroigor** [comment]
If I understand correctly, the only thing missing in the UI now is to hide/show the `Permissions` tab?

File: model/infinispan/src/main/java/org/keycloak/models/cache/infinispan/RealmAdapter.java (hunk 1 above)
Comment (vramik) [reply model/infinispan/src/main/java/org/keycloak/models/cache/infinispan/RealmAdapter.java:1852]: Makes sense, I'll consider update it.

**Comment by vramik** [comment]
> If I understand correctly, the only thing missing in the UI now is to hide/show the `Permissions` tab?

It you mean showing the `realm-management`'s authorization services, then yes. And it should be done within https://github.com/keycloak/keycloak/issues/34566. Showing a new client would be possible only after https://github.com/keycloak/keycloak/issues/34565.

**Review by pedroigor — APPROVED** [review]

**Review by agagancarczyk — APPROVED** [review]
UI changes look good!

**Pull request merged**